src/Groups/GAPGroups.jl

```diff
@@ -0,0 +1,617 @@
+  return group_element(G, x.X*y.X)
+end
+
+Base.:*(x::GAPGroupElem, y::GAPGroupElem) = _prod(x, y)
+
+function ==(x::Group, y::Group)
+   return x.X == y.X
+end
+
+function ==(x::T, y::T) where T <: GAPGroupElem
+   return x.X == y.X
+end
+
+"""
+    one(G::Group) -> x::GAPGroupElem{typeof(G)}
+Returns the one element of the group G.
+"""
+Base.one(x::Group) = group_element(x, GAP.Globals.Identity(x.X))
+
+"""
+    one(x::GAPGroupElement{T}) -> x::GAPGroupElem{T}
+Returns the one element of the parent of x.
+"""
+Base.one(x::GAPGroupElem) = one(parent(x))
+one!(x::GAPGroupElem) = one(parent(x))
+
+Base.show(io::IO, x::GAPGroupElem) =  print(io, GAP.gap_to_julia(GAP.Globals.StringView(x.X)))
+Base.show(io::IO, x::Group) = print(io, GAP.gap_to_julia(GAP.Globals.StringView(x.X)))
+
+Base.isone(x::GAPGroupElem) = x == one(parent(x))
+
+Base.inv(x::GAPGroupElem) = group_element(parent(x), GAP.Globals.Inverse(x.X))
+
+inv!(out::GAPGroupElem, x::GAPGroupElem) = inv(x)  #if needed later
+
+Base.:^(x::GAPGroupElem, y::Int) = group_element(parent(x), x.X ^ y)
+
+Base.:<(x::PermGroupElem, y::PermGroupElem) = x.X < y.X
+
+Base.:/(x::GAPGroupElem, y::GAPGroupElem) = x*y^-1
+
+mul(x::GAPGroupElem, y::GAPGroupElem) = x*y
+mul!(out::GAPGroupElem, x::GAPGroupElem, y::GAPGroupElem) = x*y
+
+div_right(x::GAPGroupElem, y::GAPGroupElem) = x*inv(y)
+div_left(x::GAPGroupElem, y::GAPGroupElem) = inv(y)*x
+div_right!(out::GAPGroupElem, x::GAPGroupElem, y::GAPGroupElem) = x*inv(y)
+div_left!(out::GAPGroupElem, x::GAPGroupElem, y::GAPGroupElem) = inv(y)*x
+
+Base.conj(x::GAPGroupElem, y::GAPGroupElem) = x^y
+Base.conj!(out::GAPGroupElem, x::GAPGroupElem, y::GAPGroupElem) = x^y
+
+comm(x::GAPGroupElem, y::GAPGroupElem) = x^-1*x^y
+comm!(out::GAPGroupElem, x::GAPGroupElem, y::GAPGroupElem) = x^-1*x^y
+
+function Base.iterate(G::Group)
+  L=GAP.Globals.Iterator(G.X)
+  if GAP.Globals.IsDoneIterator(L)
+    return nothing
+  end
+  i = GAP.Globals.NextIterator(L)
+  return group_element(G, i), L
+end
+
+function Base.iterate(G::Group, state)
+  if GAP.Globals.IsDoneIterator(state)
+    return nothing
+  end
+  i = GAP.Globals.NextIterator(state)
+  return group_element(G, i), state
+end
+
+"""
+    perm(L::Array{Int64,1})
+Given the array of positive integers `V`, returns the permutation `x` sending V[i] into V[i+1] for every i. V must contain exactly one time every integer from 1 to n for n = length(V).
+The parent of `x` is set as Sym(n), where n is the largest moved point of `x`.
+"""
+function perm(L::Array{Int64,1})
+   return PermGroupElem(symmetric_group(length(L)), GAP.Globals.PermList(GAP.julia_to_gap(L)))
+end
+
+"""
+    perm(g::PermGroup, L::Array{Int64,1})
+Given the array of positive integers `V`, returns the permutation `x` sending V[i] into V[i+1] for every i. V must contain exactly one time every integer from 1 to n for n = length(V).
+The parent of `x` is g. If `x` is not in g, it return ERROR.
+"""
+function perm(g::PermGroup, L::Array{Int64,1})
```

Comment:
the same, with the signature;
Why not `(g::PermGroup)(L::AbstractVector{<:Integer})`?

Reply:
Added it.

Reply:
if we want both, then one should dispatch to the other, the aim is to minimize code duplication;
Also, I'm not sure if the docstring is correct. Is this really what `PermList` does?

Reply:
`PermList` is a GAP function that takes in input the array `[a,b,c,d,...] ` and returns the permutation sending 1->a, 2->b, 3->c, 4->d, etc. Of course, this works only if the array contains exactly once every number from 1 to `n` for some integer `n`. In my code, this function is renamed as `perm`.

Reply:
yes, this is what I thought it does. But this is not what the docstring says?!

Also: there is already one `perm(::AbstractVector{<:Integer})` function in `Oscar.jl`, so the definition above overrides it. we shouldn't do it